ports/espressif/bindings/espnow/ESPNow.c

```diff
@@ -0,0 +1,876 @@
+
+//|     def del_peer(self, mac: bytes) -> None:
+//|         """Delete peer.
+//|
+//|         :param bytes mac: The mac address of the peer."""
+//|         ...
+STATIC mp_obj_t espnow_del_peer(mp_obj_t self_in, mp_obj_t mac) {
+    espnow_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    check_for_deinit(self);
+
+    uint8_t peer_addr[ESP_NOW_ETH_ALEN];
+    memcpy(peer_addr, _get_peer_addr(mac), ESP_NOW_ETH_ALEN);
+
+    check_esp_err(esp_now_del_peer(peer_addr));
+    _update_peer_count(self);
+
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_2(espnow_del_peer_obj, espnow_del_peer);
+
+// Convert a peer_info struct to python tuple
+// Used by espnow_get_peer() and espnow_get_peers()
+static mp_obj_t _peer_info_to_tuple(const esp_now_peer_info_t *peer) {
+    return MP_OBJ_NEW_TUPLE(
+        mp_obj_new_bytes(peer->peer_addr, MP_ARRAY_SIZE(peer->peer_addr)),
+        mp_obj_new_bytes(peer->lmk, MP_ARRAY_SIZE(peer->lmk)),
+        mp_obj_new_int(peer->channel),
+        mp_obj_new_int(peer->ifidx),
+        mp_obj_new_bool(peer->encrypt));
+}
+
+//|     def get_peer(self, mac: bytes) -> Tuple[bytes, int, int, bool]:
+//|         """Get the peer info for mac as a `tuple`.
+//|
+//|         :param bytes mac: The mac address of the peer.
+//|
+//|         :returns: A `tuple` of (mac, lmk, channel, interface, encrypt)."""
+//|         ...
+STATIC mp_obj_t espnow_get_peer(mp_obj_t self_in, mp_obj_t mac) {
+    espnow_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    check_for_deinit(self);
+
+    esp_now_peer_info_t peer = {0};
+    memcpy(peer.peer_addr, _get_peer_addr(mac), ESP_NOW_ETH_ALEN);
+    check_esp_err(esp_now_get_peer(peer.peer_addr, &peer));
+
+    return _peer_info_to_tuple(&peer);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_2(espnow_get_peer_obj, espnow_get_peer);
+
+// --- Peer Related Properties ---
+
+//|     peers: Tuple[Tuple[bytes, bytes, int, int, bool], ...]
+//|     """The peer info records for all registered `ESPNow` peers. (read-only)
+//|
+//|     A `tuple` of tuples: ((mac, lmk, channel, interface, encrypt), ...)."""
+//|
+STATIC mp_obj_t espnow_get_peers(mp_obj_t self_in) {
+    espnow_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    check_for_deinit(self);
+
+    // Build and initialize the peer info tuple.
+    mp_obj_tuple_t *peerinfo_tuple = mp_obj_new_tuple(self->peers_count, NULL);
+    esp_now_peer_info_t peer = {0};
+
+    for (size_t i = 0; i < peerinfo_tuple->len; i++) {
+        esp_err_t status = esp_now_fetch_peer((i == 0), &peer);
+        peerinfo_tuple->items[i] = (status == ESP_OK ? _peer_info_to_tuple(&peer) : mp_const_none);
+    }
+
+    return peerinfo_tuple;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(espnow_get_peers_obj, espnow_get_peers);
+
+MP_PROPERTY_GETTER(espnow_peers_obj,
+    (mp_obj_t)&espnow_get_peers_obj);
+
+//|     peers_count: Tuple[int, int]
+//|     """The number of registered peers in a `tuple` of (num_total_peers, num_encrypted_peers). (read-only)"""
```

Comment:
Is it important to get these two values at the same time? Otherwise, I'd split them into two properties.